reference/variables.md

```diff
@@ -228,6 +248,48 @@ paths:
 | https://example.com/api/bar | /** | /api/bar | /api |
 
 
+## `upstream`
+
+The `$upstream` variable contains information about upstream responses. The properties for each upstream response are stored with the request ID ([`id` property](/reference/actions/request.md#id) or [`content` attribute](/reference/actions/request.md#syntax)).
+
+* `url` - The request URL (string)
+* `status` - The response status code (integer)
+* `cacheHit` - `true` if the response was served from a cache (see [`use-http-cache` or `force-cache-ttl` request options](/reference/actions/request.md#options)), `false` otherwise
+* ``headers - The response headers, each with a lower-cased field name
+
+Example:
+```xml
+<upstream>
+  <ok>
+    <url>https://httpbin.org/status/200</url>
+    <status number="">200</status>
+    <cacheHit>false</cacheHit>
+    <headers>
+      <date>Thu, 27 Aug 2020 14:12:33 GMT</date>
+      <content-type>text/html; charset=utf-8</content-type>
+      <connection>keep-alive</connection>
+      <server>gunicorn/19.9.0</server>
+      <access-control-allow-origin>*</access-control-allow-origin>
+      <access-control-allow-credentials>true</access-control-allow-credentials>
+    </headers>
+  </ok>
+  <failure>
```

Comment:
`ok` and `failure` may be misleading examples. this is the content-id? what about `request1` and `request2`? or `id1` and `id2`?

Reply:
This is the result from the example flow in lines 47ff.

Reply:
ah, missed that in the diff view. than "ok" :)